applications/StructuralMechanicsApplication/custom_elements/updated_lagrangian.h

```diff
@@ -152,8 +230,10 @@ class UpdatedLagrangian
     ///@name Protected member Variables
     ///@{
 
-    std::vector<double> mDetF0; // Historical total elastic deformation measure determinant
-    std::vector<Matrix> mF0; // Historical total elastic deformation measure
+    /* Historical total elastic deformation measure */
+    bool mF0Computed;           // To avoid computing more than once the historical total elastic deformation measure 
+    std::vector<double> mDetF0; // The historical total elastic deformation measure determinant
```

Comment:
@loumalouomega, what exactly are u storing in F0? 
I am asking this since a UL should store in there something like Dx(n)/Dx0. There are other alternatives though, so please specify it.

@iiaconeta i think i would approve this push request, however i also believe we should be pending on you to revise the formulation to see if it fits in what you need. please confirm if you can (or can not) do it